File: Leetcode/UniqueBinarySearchTree/DP.cpp
class Solution {
  public:
    int numTrees(int n) {
      vector<int> f(n + 1, 0);
      f[0] = 1;
      f[1] = 1;

      for(int i = 2; i <= n; i++){
        for(int j = 0; j < i; j++){
          f[i] += f[j] * f[i - j - 1];
        }
      }
      return f[n];
    }
};

File: Leetcode/UniquePath/DP.py
class Solution(object):
    def uniquePaths(self, m, n):
        if m == 0 or n == 0:
            return 0
        mat = list();
        for i in range(0, m):
            v = [0] * n
            mat.append(v)
        
        mat[0][0] = 1
        
        for i in range(0, m):
            for j in range(0, n):
                if i - 1 >= 0:
                    mat[i][j] += mat[i-1][j]
                if j - 1 >= 0:
                    mat[i][j] += mat[i][j-1]
        return mat[m-1][n-1]
        
